libpd_wrapper/util/z_queued.c

```diff
@@ -12,23 +12,32 @@
 
 #include <stdlib.h>
 #include <string.h>
-
+#include "z_hooks.h"
 #include "ringbuffer.h"
+#include "z_print_util.h"
+
+#define BUFFER_SIZE 16384
 
-t_libpd_printhook libpd_queued_printhook = NULL;
-t_libpd_banghook libpd_queued_banghook = NULL;
-t_libpd_floathook libpd_queued_floathook = NULL;
-t_libpd_symbolhook libpd_queued_symbolhook = NULL;
-t_libpd_listhook libpd_queued_listhook = NULL;
-t_libpd_messagehook libpd_queued_messagehook = NULL;
-
-t_libpd_noteonhook libpd_queued_noteonhook = NULL;
-t_libpd_controlchangehook libpd_queued_controlchangehook = NULL;
-t_libpd_programchangehook libpd_queued_programchangehook = NULL;
-t_libpd_pitchbendhook libpd_queued_pitchbendhook = NULL;
-t_libpd_aftertouchhook libpd_queued_aftertouchhook = NULL;
-t_libpd_polyaftertouchhook libpd_queued_polyaftertouchhook = NULL;
-t_libpd_midibytehook libpd_queued_midibytehook = NULL;
+typedef struct _queued_stuff {
+  t_libpd_printhook q_printhook;
```

Comment:
I was following the idea of allocating only needed memory; like when the user doesn't need queuing but targets a strongly constrained system. Do you think the saving is negligible?

Reply:
Well I was mainly thinking of the 16kB buffer, but it could be now allocated dynamically by `libpd_queued_stuff_new()`.

Do you think all hooks related stuff should move to `z_libpdhooks`, including the content of current  `t_libpd_concatenated_print_stuff` (in `util/z_print_util.c`)?

It would result in a all-in-one z_libpdhooks:
``` C
typedef struct _libpdhooks {

  // messages
  // no h_printhook as libpd_set_printhook() sets internal STUFF->st_printhook
  t_libpd_banghook h_banghook;
  t_libpd_floathook h_floathook;
  t_libpd_symbolhook h_symbolhook;
  t_libpd_listhook h_listhook;
  t_libpd_messagehook h_messagehook;

  // MIDI
  t_libpd_noteonhook h_noteonhook;
  t_libpd_controlchangehook h_controlchangehook;
  t_libpd_programchangehook h_programchangehook;
  t_libpd_pitchbendhook h_pitchbendhook;
  t_libpd_aftertouchhook h_aftertouchhook;
  t_libpd_polyaftertouchhook h_polyaftertouchhook;
  t_libpd_midibytehook h_midibytehook;
  
  // concatenation stuff
  char *h_concatenated_printbuffer;
  int h_concatenated_printbuflen;
  t_libpd_printhook c_concatenated_printhook;
  
  // queued stuff
  t_libpd_printhook h_queued_printhook;
  t_libpd_banghook h_queued_banghook;
  t_libpd_floathook h_queued_floathook;
  t_libpd_symbolhook h_queued_symbolhook;
  t_libpd_listhook h_queued_listhook;
  t_libpd_messagehook h_queued_messagehook;

  t_libpd_noteonhook h_queued_noteonhook;
  t_libpd_controlchangehook h_queued_controlchangehook;
  t_libpd_programchangehook h_queued_programchangehook;
  t_libpd_pitchbendhook h_queued_pitchbendhook;
  t_libpd_aftertouchhook h_queued_aftertouchhook;
  t_libpd_polyaftertouchhook h_queued_polyaftertouchhook;
  t_libpd_midibytehook h_queued_midibytehook;

  ring_buffer *h_queued_pd_receive_buffer;
  ring_buffer *h_queued_midi_receive_buffer;
  char *h_queued_buffer;

} t_libpdhooks;
```

Reply:
ping @danomatika

Reply:
done.